### ISSUE TITLE
`redundant_closure` false positive when mapping to enum variant

### ISSUE THREAD

**Issue by connorskees** [rust-lang/rust-clippy#5126]
Hello!

I don't believe there is a duplicate issue for this. Clippy will incorrectly warn when attempting to map an item to an enum:

```rust
enum Foo {
    X(i32),
}

fn main() {
    let _: Vec<Foo> = vec![0, 1, 2].into_iter().map(|y| Foo::X(y)).collect();
    // this fails to compile
    // let _: Vec<Foo> = vec![0, 1, 2].into_iter().map(|y| Foo::X).collect();
}
```

```
warning: redundant closure found
 --> src\main.rs:6:53
  |
6 |     let _: Vec<Foo> = vec![0, 1, 2].into_iter().map(|y| Foo::X(y)).collect();
  |                                                     ^^^^^^^^^^^^^ help: remove closure as shown: `Foo::X`
  |
  = note: `#[warn(clippy::redundant_closure)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure
```

```
cargo clippy -V
clippy 0.0.212 (c0f39cfb 2020-01-29)
```

**Comment by Areredify** [comment]
It is intended to be 
```
let _: Vec<Foo> = vec![0, 1, 2].into_iter().map(Foo::X).collect();
```
Each enum variant has a corresponding constructor, which you can use as a parameter in `map`.

**Comment by connorskees** [comment]
Ah my bad. Thank you!
